src/lib/y2partitioner/widgets/summary_text.rb

```diff
@@ -100,9 +100,7 @@ def calculate_actions
 
       # Updates the value of {#packages}
       def calculate_packages
-        handler = Y2Storage::PackageHandler.new
-        handler.add_feature_packages(current_graph)
-        @packages = handler.pkg_list.uniq
+        @packages = current_graph.used_features.pkg_list
```

Comment:
AFAIS, `#calculate_packages` is only called in the `#init` method, right? Why not simply replace this method with a `#packages` one:

```ruby
def packages
  # feel free to remove memoization if it is not needed
  @packages ||= current_graph.used_features.pkg_list
end
```

You could drop the `attr_reader :packages` and simplify the `#init` method.

Reply:
All that is improved in a subsequent commit. This whole API made very little sense and this pull request was just intended as an initial minimal polishing (which is exactly the name of the commit).